public/app/features/browse-dashboards/state/hooks.ts

```diff
@@ -32,6 +32,16 @@ const selectedItemsForActionsSelector = createSelector(
   }
 );
 
+export function useBrowseLoadingStatus(folderUID: string | undefined): 'pending' | 'fulfilled' {
+  return useSelector((wholeState) => {
+    const children = folderUID
+      ? wholeState.browseDashboards.childrenByParentUID[folderUID]
+      : wholeState.browseDashboards.rootItems;
+
+    return children ? 'fulfilled' : 'pending';
+  });
+}
```

Comment:
Can we get a test for this? I can accept it in another PR if this needs to be merged ASAP